src/web/components/SignInGate/SignInGateSelector.tsx

```diff
@@ -144,6 +144,38 @@ const ShowSignInGate = ({
 	return <></>;
 };
 
+// custom hook can be used by other parts of DCR to determine if the sign in gate will or will not show
+export const useSignInGateWillShow = ({
```

Comment:
Most custom hooks in DCR are stored in their own files in the /libs folder. There's a kind of vague convention of one file one export plus it maybe helps with discoverability and refactoring later

Reply:
Great, I'll move the hook into it's own file in the web/lib folder